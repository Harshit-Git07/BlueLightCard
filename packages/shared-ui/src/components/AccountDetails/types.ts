export type AccountDetailsProps = {
  accountNumber: string;
  firstName: string;
  lastName: string;
};
