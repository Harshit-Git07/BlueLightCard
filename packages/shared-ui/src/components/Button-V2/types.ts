import { MouseEventHandler, PropsWithChildren } from 'react';
import { IconDefinition } from '@fortawesome/fontawesome-svg-core';
import { ThemeVariant } from '../../types';

export type ButtonProps = PropsWithChildren & {
  className?: string;
  type?: 'button' | 'submit';
  disabled?: boolean;
  slim?: boolean;
  variant?: ThemeVariant;
  invertColor?: boolean;
  iconLeft?: IconDefinition;
  iconRight?: IconDefinition;
  href?: string;
  newTab?: boolean;
  onClick?: MouseEventHandler;
  withoutHover?: boolean;
  withoutFocus?: boolean;
  borderless?: boolean;
  size?: ButtonSize;
  'data-testid'?: string;
};

export type ButtonSize = 'XSmall' | 'Small' | 'Large';
