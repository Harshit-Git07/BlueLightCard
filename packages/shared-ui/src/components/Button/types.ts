import { MouseEventHandler, PropsWithChildren } from 'react';
import { IconDefinition } from '@fortawesome/fontawesome-svg-core';
import { ThemeVariant } from '../../types';

export type ColorToken = {
  bg: string;
  hover: string;
  focus: string;
  text: string;
  border: string;
};

export type ButtonProps = PropsWithChildren & {
  type?: 'button' | 'submit' | 'link';
  disabled?: boolean;
  slim?: boolean;
  className?: string;
  variant?: ThemeVariant;
  invertColor?: boolean;
  iconLeft?: IconDefinition;
  iconRight?: IconDefinition;
  href?: string;
  onClick?: MouseEventHandler;
  withoutHover?: boolean;
  withoutFocus?: boolean;
  borderless?: boolean;
};
