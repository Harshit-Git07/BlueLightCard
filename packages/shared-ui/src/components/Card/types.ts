import { IconDefinition } from '@fortawesome/fontawesome-svg-core';
import { RefObject } from 'react';

export type CardState = 'default' | 'hover' | 'selected';

export interface CardProps {
  initialCardState?: CardState;
  cardTitle?: string;
  description?: string;
  truncateDescription?: boolean
  buttonTitle?: string;
  imageAlt?: string;
  imageSrc?: string;
  iconLeft?: IconDefinition;
  iconRight?: IconDefinition;
  showImage?: boolean;
  showDescription?: boolean;
  showButton?: boolean;
  onClick?: () => void;
  ariaLabel?: string;
  className?: string;
  'data-testid'?: string;
  canHover?: boolean;
}

export interface CardContentProps {
  title: string;
  description?: string;
  showDescription?: boolean;
  centerContent?: boolean;
  ariaLabel?: string;
  descriptionRef: RefObject<HTMLParagraphElement>;
  truncateDescription?: boolean;
}
