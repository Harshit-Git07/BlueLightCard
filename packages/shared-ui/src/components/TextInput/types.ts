import { ChangeEventHandler, KeyboardEventHandler } from 'react';

export type TextInputState = 'Default' | 'Active' | 'Filled' | 'Error' | 'Disabled' | 'Editable';

export interface TextInputProps {
  id?: string;
  name?: string;
  state: TextInputState;
  value?: string;
  required?: boolean;
  maxChars?: number;
  onChange?: ChangeEventHandler<HTMLInputElement>;
  onKeyDown?: KeyboardEventHandler<HTMLInputElement>;
  placeholder?: string;
  isEditable?: boolean;
  maxlength?: number;
  showCharCount?: boolean;
  min?: number;
  max?: number;
  label: string;
  showLabel?: boolean;
  showIcon?: boolean;
  helpMessage?: string;
  showHelpMessage?: boolean;
  infoMessage?: string;
  showInfoMessage?: boolean;
  ariaLabel?: string;
}
