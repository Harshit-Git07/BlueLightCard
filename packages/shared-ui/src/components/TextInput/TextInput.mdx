import { Canvas, Controls, Meta } from '@storybook/blocks'
import * as TextInput from './TextInput.stories'
import ComponentStatus from '../../../.storybook/blocks/ComponentStatus'

<Meta of={TextInput}/>

# TextInput

- [Status](#status)
- [Overview](#overview)
- [Props](#props)
- [State](#state)
- [Variants and States](#variants-and-states)
- [Implementation](#implementation)
- [Usage Guidelines](#usage-guidelines)
- [Design](#design)

## Status
<ComponentStatus/>

## Overview

The Text Input component is a versatile UI element used to create text field input areas in forms. It typically includes a label and optional helper text or validation messages. This component ensures consistency and accessibility in form design, providing a structured and user-friendly way to capture user input.

## Props
<Canvas of={TextInput.Default}/>
<Controls/>


## Variants and States

List of variants can be shown below here.

**Default**
<Canvas of={TextInput.Default}/>

**Filled**
<Canvas of={TextInput.Filled}/>

**Error**
<Canvas of={TextInput.Error}/>

**Success**
<Canvas of={TextInput.Success}/>

**Disabled**
<Canvas of={TextInput.Disabled}/>

## Implementation
Usage example using jsx

```jsx
import React, { useState } from 'react';
import TextInput from './TextInput';

const ExampleComponent = () => {
  const [value, setValue] = useState(args.value ?? '');
  const [validationMessage, setValidationMessage] = useState(args.validationMessage);

  const onChange = (e: ChangeEvent<HTMLInputElement>) => {
    const newValue = e.target.value;
    setValue(newValue.length <= args.maxLength ? newValue : value);
  };

  useEffect(() => {
    if (args.maxLength) {
      const charsLeft = args.maxLength - value.length;
      const charsLeftString = `${charsLeft} characters remaining`;
      setValidationMessage(charsLeftString);
    }
  }, [value]);

  return (
    <TextInput {...args} value={value} validationMessage={validationMessage} onChange={onChange} />
  );
};

export default ExampleComponent;
```

## Usage Guidelines

➡️ Always use placeholder in input field where indicated

➡️ Character or word counters should be used if there is a character or word limit.

## Design
- [Confluence: Text Input Documentation](https://bluelightcard.atlassian.net/wiki/spaces/PE/pages/2935881730/Text+Input)
- [Figma: TextInput](https://www.figma.com/design/n2NzB4G3OTR74o3dQ4BOwm/My-Account-%26-Sign-up-Component-library?node-id=22-4300&node-type=canvas&m=dev)

