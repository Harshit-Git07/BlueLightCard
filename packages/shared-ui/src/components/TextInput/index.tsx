import React, { FC, useEffect, useState } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCircleInfo } from '@fortawesome/pro-solid-svg-icons';

import { TextInputProps, TextInputState } from './types';

const getInputClasses = (state: TextInputState) => {
  const baseClasses = 'w-full rounded-md px-4 border focus:outline-none pt-6 pb-2';

  switch (state) {
    case 'Active':
    case 'Filled':
      return `${baseClasses} border-colour-primary dark:border-colour-primary-dark bg-transparent`;
    case 'Error':
      return `${baseClasses} border-colour-error dark:border-colour-error bg-transparent`;
    case 'Disabled':
      return `${baseClasses} bg-colour-surface-container border-colour-onSurface-outline-subtle dark:bg-colour-surface-container-dark dark:border-colour-onSurface-outline-subtle-dark`;
    default:
      return `${baseClasses} border-colour-onSurface-outline dark:border-colour-onSurface-outline-dark bg-transparent`;
  }
};

const getPlaceholderClasses = (state: TextInputState) => {
  const baseClasses = 'absolute left-4 transition-all duration-200 pointer-events-none';
  const activeClasses =
    'font-typography-label font-typography-label-weight text-typography-label leading-typography-label top-2';
  const inactiveClasses =
    'font-typography-body font-typography-body-weight text-typography-body leading-typography-body top-1/2 -translate-y-1/2';

  switch (state) {
    case 'Active':
    case 'Filled':
      return `${baseClasses} ${activeClasses} text-colour-onSurface-subtle dark:text-colour-onSurface-subtle-dark`;
    case 'Disabled':
      return `${baseClasses} ${inactiveClasses} text-colour-onSurface-disabled dark:text-colour-onSurface-disabled-dark`;
    default:
      return `${baseClasses} ${inactiveClasses} text-colour-onSurface-subtle dark:text-colour-onSurface-subtle-dark`;
  }
};

const getMessageLabelClasses = (state: TextInputState) => {
  switch (state) {
    case 'Active':
      return 'text-colour-primary-light dark:text-colour-primary-dark';
    case 'Error':
      return 'text-colour-error dark:text-colour-error-dark';
    default:
      return 'text-colour-onSurface-subtle dark:text-colour-onSurface-subtle-dark';
  }
};

const TextInput: FC<TextInputProps> = ({
  id,
  name,
  state = 'Default',
  value,
  required = false,
  maxChars = 200,
  showCharCount,
  onChange,
  onKeyDown,
  placeholder,
  min,
  max,
  label,
  showLabel = false,
  showIcon = false,
  helpMessage,
  showHelpMessage = false,
  infoMessage,
  showInfoMessage = false,
  ariaLabel,
}) => {
  const [inputState, setInputState] = useState<TextInputState>(state);
  const remainingChars = maxChars - (value?.length ?? 0);

  useEffect(() => {
    setInputState(state);
  }, [state]);

  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const newValue = e.target.value;
    if (newValue.length <= maxChars) {
      if (onChange) {
        onChange(e);
      }
    }
  };

  const handleFocus = () => {
    setInputState('Active');
  };

  const handleBlur = () => {
    if (value && value.length > 0) {
      setInputState('Filled');
    } else {
      setInputState(state);
    }
  };

  const helpMessageId = `${name}-help`;
  const infoMessageId = `${name}-info`;
  const charCountId = `${name}-char-count`;

  const getAriaDescribedBy = () => {
    const ids = [];
    if (showHelpMessage && helpMessage) ids.push(helpMessageId);
    if (showInfoMessage && infoMessage) ids.push(infoMessageId);
    if (showCharCount) ids.push(charCountId);
    return ids.length > 0 ? ids.join(' ') : undefined;
  };

  return (
    <div className="space-y-2">
      {showLabel && (
        <div className="flex items-center gap-2">
          {showLabel && (
            <label
              htmlFor={id ?? name}
              aria-label={ariaLabel}
              className="text-colour-onSurface dark:text-colour-onSurface-dark font-typography-body text-typography-body font-typography-body-weight leading-typography-body tracking-typography-body"
            >
              {label}
            </label>
          )}
          {showIcon && showLabel && (
            <FontAwesomeIcon
              className="text-colour-onSurface-subtle dark:text-colour-onSurface-subtle-dark"
              icon={faCircleInfo}
            />
          )}
        </div>
      )}

      {showHelpMessage && helpMessage && (
        <div
          id={helpMessageId}
          className="text-colour-onSurface-subtle dark:text-colour-onSurface-subtle-dark font-typography-body text-typography-body font-typography-body-weight leading-typography-body tracking-typography-body"
        >
          {helpMessage}
        </div>
      )}

      <div className="relative h-[50px]">
        <input
          id={id ?? name}
          className={`${getInputClasses(inputState)} text-colour-onSurface dark:text-colour-onSurface-dark h-full`}
          value={value}
          maxLength={maxChars}
          min={min}
          max={max}
          name={name}
          required={required}
          onChange={handleChange}
          onKeyDown={onKeyDown}
          onFocus={handleFocus}
          onBlur={handleBlur}
          disabled={inputState === 'Disabled'}
          aria-invalid={state === 'Error'}
          aria-required={required}
          aria-describedby={getAriaDescribedBy()}
        />
        <span className={getPlaceholderClasses(inputState)}>{placeholder}</span>
      </div>

      {showInfoMessage && infoMessage && (
        <div>
          <label
            id={infoMessageId}
            className={`font-typography-body-small font-typography-body-small-weight text-typography-body-small leading-typography-body-small ${getMessageLabelClasses(inputState)}`}
          >
            {infoMessage}
          </label>
        </div>
      )}
      {showCharCount && (
        <div
          id={charCountId}
          className={`font-typography-body-small font-typography-body-small-weight text-typography-body-small leading-typography-body-small ${getMessageLabelClasses(inputState)}`}
        >
          {remainingChars} characters remaining
        </div>
      )}
    </div>
  );
};

export default TextInput;
