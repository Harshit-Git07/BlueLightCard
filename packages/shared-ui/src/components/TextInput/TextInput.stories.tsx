import React from 'react';
import { Meta, StoryFn } from '@storybook/react';
import TextInput from './';
import { TextInputState } from './types';

const meta: Meta<typeof TextInput> = {
  title: 'Component System/TextInput',
  component: TextInput,
  parameters: {
    status: 'wip',
  },
  argTypes: {
    state: {
      control: 'select',
      options: ['Default', 'Active', 'Filled', 'Error', 'Disabled'],
    },
    showLabel: {
      control: 'boolean',
    },
    showIcon: {
      control: 'boolean',
    },
    required: {
      control: 'boolean',
    },
    showHelpMessage: {
      control: 'boolean',
    },
    showInfoMessage: {
      control: 'boolean',
    },
  },
};

export default meta;

const Template: StoryFn<typeof TextInput> = (args) => {
  const [value, setValue] = React.useState(args.value ?? '');
  return <TextInput {...args} value={value} onChange={(e) => setValue(e.target.value)} />;
};

export const Default = Template.bind({});
Default.args = {
  name: 'default-input',
  placeholder: 'Enter text here',
  label: 'Default Input',
  ariaLabel: 'This is a example aria text for screen readers',
  state: 'Default' as TextInputState,
  showLabel: true,
};

export const Active = Template.bind({});
Active.args = {
  ...Default.args,
  state: 'Active' as TextInputState,
};

export const Filled = Template.bind({});
Filled.args = {
  ...Default.args,
  state: 'Filled' as TextInputState,
  value: 'This is a filled input',
};

export const Error = Template.bind({});
Error.args = {
  ...Default.args,
  state: 'Error' as TextInputState,
  required: true,
  infoMessage: 'This field is required',
  showInfoMessage: true,
};

export const Disabled = Template.bind({});
Disabled.args = {
  ...Default.args,
  state: 'Disabled' as TextInputState,
};

export const ShowLabelWithIcon = Template.bind({});
ShowLabelWithIcon.args = {
  ...Default.args,
  showLabel: true,
};

export const ShowLabelWithoutIcon = Template.bind({});
ShowLabelWithoutIcon.args = {
  ...Default.args,
  showIcon: true,
  helpMessage: 'This is a help message',
};

export const ShowHelpMessage = Template.bind({});
ShowHelpMessage.args = {
  ...Default.args,
  helpMessage: 'This is a help message',
  showHelpMessage: true,
};

export const ShowInfoMessage = Template.bind({});
ShowInfoMessage.args = {
  ...Default.args,
  infoMessage: 'Additional information about this field',
  showInfoMessage: true,
};

export const WithMaxChars = Template.bind({});
WithMaxChars.args = {
  ...Default.args,
  maxChars: 20,
  showCharCount: true,
};

export const WithAllFeatures = Template.bind({});
WithAllFeatures.args = {
  name: 'all-features-input',
  placeholder: 'Enter text here',
  label: 'Input with All Features',
  state: 'Default' as TextInputState,
  showLabel: true,
  showIcon: true,
  required: true,
  maxChars: 50,
  helpMessage: 'This is a help message',
  showHelpMessage: true,
  infoMessage: 'Additional information about this field',
  showInfoMessage: true,
  ariaLabel: 'This is a example aria text for screen readers',
  showCharCount: true,
};
