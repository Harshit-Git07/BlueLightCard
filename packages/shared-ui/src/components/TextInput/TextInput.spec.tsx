import TextInput from './';
import { TextInputProps } from './types';
import { colours } from '../../tailwind/theme';
import userEvent from '@testing-library/user-event';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';

const mockOnChange = jest.fn();
const mockOnKeyDown = jest.fn();

const setup = (partialProps?: Partial<TextInputProps>) => {
  const defaultProps: TextInputProps = {
    id: 'test-input',
    name: 'test',
    isValid: undefined,
    isDisabled: false,
    value: '',
    isRequired: false,
    onChange: mockOnChange,
    onKeyDown: mockOnKeyDown,
    placeholder: 'Enter text',
    label: 'Test Label',
    tooltip: 'Helpful tooltip',
    description: 'Field description',
    validationMessage: 'Validation message',
  };

  return render(<TextInput {...defaultProps} {...partialProps} />);
};

describe('TextInput component', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('renders the label when provided', () => {
    setup();
    expect(screen.getByText('Test Label')).toBeInTheDocument();
  });

  it('does not render the label when not provided', () => {
    setup({ label: undefined });
    expect(screen.queryByText('Test Label')).not.toBeInTheDocument();
  });

  it('renders the tooltip and description when provided', () => {
    setup();
    expect(screen.getByText('Helpful tooltip')).toBeInTheDocument();
    expect(screen.getByText('Field description')).toBeInTheDocument();
  });

  it('renders the floating placeholder', () => {
    setup();
    expect(screen.getByText('Enter text')).toBeInTheDocument();
  });

  it('renders the validation message when provided', () => {
    setup();
    expect(screen.getByText('Validation message')).toBeInTheDocument();
  });

  it('applies disabled styles and attributes when isDisabled is true', () => {
    setup({ isDisabled: true });
    const input = screen.getByRole('textbox');
    expect(input).toBeDisabled();
    expect(input).toHaveClass(colours.backgroundSurfaceContainer);
  });

  it('applies aria-invalid when isValid is false', () => {
    setup({ isValid: false });
    const input = screen.getByRole('textbox');
    expect(input).toHaveAttribute('aria-invalid', 'true');
  });

  it('does not apply aria-invalid when isValid is true', () => {
    setup({ isValid: true });
    const input = screen.getByRole('textbox');
    expect(input).toHaveAttribute('aria-invalid', 'false');
  });

  it('applies aria-required when isRequired is true', () => {
    setup({ isRequired: true });
    const input = screen.getByRole('textbox');
    expect(input).toHaveAttribute('aria-required', 'true');
  });

  it('triggers onChange when input value changes', () => {
    setup();
    const input = screen.getByRole('textbox');
    fireEvent.change(input, { target: { value: 'New Value' } });
    expect(mockOnChange).toHaveBeenCalled();
  });

  it('triggers onKeyDown when a key is pressed', () => {
    setup();
    const input = screen.getByRole('textbox');
    fireEvent.keyDown(input, { key: 'Enter', code: 'Enter' });
    expect(mockOnKeyDown).toHaveBeenCalled();
  });

  it('toggles focus state when input gains and loses focus', async () => {
    setup();
    const input = screen.getByRole('textbox');

    userEvent.click(input);
    await waitFor(() => {
      expect(input).toHaveFocus();
    });

    userEvent.tab();
    await waitFor(() => {
      expect(input).not.toHaveFocus();
    });
  });

  it('uses the correct aria-describedby when descriptive components are provided', () => {
    setup();
    const input = screen.getByRole('textbox');
    expect(input).toHaveAttribute(
      'aria-describedby',
      'test-input-tooltip test-input-description test-input-placeholder test-input-validation-message',
    );
  });

  it('does not set aria-describedby when all descriptive components are absent', () => {
    setup({
      tooltip: undefined,
      description: undefined,
      validationMessage: undefined,
      label: undefined,
      placeholder: undefined,
    });
    const input = screen.getByRole('textbox');
    expect(input).toHaveAttribute('aria-describedby', '');
  });
});
