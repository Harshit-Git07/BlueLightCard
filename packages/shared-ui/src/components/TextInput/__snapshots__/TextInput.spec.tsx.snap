// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput component snapshot tests renders default input correctly 1`] = `
<div
  className="space-y-2"
>
  <div
    className="relative h-[50px]"
  >
    <input
      aria-invalid={false}
      aria-required={false}
      className="w-full rounded-md px-4 border focus:outline-none pt-6 pb-2 border-colour-onSurface-outline dark:border-colour-onSurface-outline-dark bg-transparent text-colour-onSurface dark:text-colour-onSurface-dark h-full"
      disabled={false}
      id="test-input"
      maxLength={200}
      name="test-input"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      value=""
    />
    <span
      className="absolute left-4 transition-all duration-200 pointer-events-none font-typography-body font-typography-body-weight text-typography-body leading-typography-body top-1/2 -translate-y-1/2 text-colour-onSurface-subtle dark:text-colour-onSurface-subtle-dark"
    >
      Enter text
    </span>
  </div>
</div>
`;

exports[`TextInput component snapshot tests renders input in error state 1`] = `
<div
  className="space-y-2"
>
  <div
    className="relative h-[50px]"
  >
    <input
      aria-describedby="test-input-info"
      aria-invalid={true}
      aria-required={false}
      className="w-full rounded-md px-4 border focus:outline-none pt-6 pb-2 border-colour-error dark:border-colour-error bg-transparent text-colour-onSurface dark:text-colour-onSurface-dark h-full"
      disabled={false}
      id="test-input"
      maxLength={200}
      name="test-input"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      value=""
    />
    <span
      className="absolute left-4 transition-all duration-200 pointer-events-none font-typography-body font-typography-body-weight text-typography-body leading-typography-body top-1/2 -translate-y-1/2 text-colour-onSurface-subtle dark:text-colour-onSurface-subtle-dark"
    >
      Enter text
    </span>
  </div>
  <div>
    <label
      className="font-typography-body-small font-typography-body-small-weight text-typography-body-small leading-typography-body-small text-colour-error dark:text-colour-error-dark"
      id="test-input-info"
    >
      Error message
    </label>
  </div>
</div>
`;

exports[`TextInput component snapshot tests renders input with label and icon 1`] = `
<div
  className="space-y-2"
>
  <div
    className="flex items-center gap-2"
  >
    <label
      className="text-colour-onSurface dark:text-colour-onSurface-dark font-typography-body text-typography-body font-typography-body-weight leading-typography-body tracking-typography-body"
      htmlFor="test-input"
    >
      Test-label
    </label>
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-circle-info text-colour-onSurface-subtle dark:text-colour-onSurface-subtle-dark"
      data-icon="circle-info"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={{}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM216 336h24V272H216c-13.3 0-24-10.7-24-24s10.7-24 24-24h48c13.3 0 24 10.7 24 24v88h8c13.3 0 24 10.7 24 24s-10.7 24-24 24H216c-13.3 0-24-10.7-24-24s10.7-24 24-24zm40-208a32 32 0 1 1 0 64 32 32 0 1 1 0-64z"
        fill="currentColor"
        style={{}}
      />
    </svg>
  </div>
  <div
    className="relative h-[50px]"
  >
    <input
      aria-invalid={false}
      aria-required={false}
      className="w-full rounded-md px-4 border focus:outline-none pt-6 pb-2 border-colour-onSurface-outline dark:border-colour-onSurface-outline-dark bg-transparent text-colour-onSurface dark:text-colour-onSurface-dark h-full"
      disabled={false}
      id="test-input"
      maxLength={200}
      name="test-input"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      value=""
    />
    <span
      className="absolute left-4 transition-all duration-200 pointer-events-none font-typography-body font-typography-body-weight text-typography-body leading-typography-body top-1/2 -translate-y-1/2 text-colour-onSurface-subtle dark:text-colour-onSurface-subtle-dark"
    >
      Enter text
    </span>
  </div>
</div>
`;
